builder_test.go

```diff
@@ -34,14 +34,14 @@ func TestBuilder(t *testing.T) {
 }
 
 //go:generate mockgen -package testmock -destination testmock/env.go github.com/buildpacks/lifecycle BuildEnv
-//go:generate mockgen -package testmock -destination testmock/buildpack_store.go github.com/buildpacks/lifecycle BuildpackStore
-//go:generate mockgen -package testmock -destination testmock/buildpack.go github.com/buildpacks/lifecycle Buildpack
+//go:generate mockgen -package testmock -destination testmock/dir_store.go github.com/buildpacks/lifecycle DirStore
+//go:generate mockgen -package testmock -destination testmock/build_module.go github.com/buildpacks/lifecycle/buildpack BuildModule
```

Comment:
"Build module" seemed the best way of describing something that can be a buildpack or an extension. Open to other suggestions.

Reply:
Thought about this a little...best I could come up with was `DetectorBuilder` (similar in spirit to the standard library `ReadCloser`, for example).
Otherwise, maybe something like `GeneralizedBuildpack`.
Thought about `Builder`, but that's a different thing in our world.

Not strongly convinced any of that is better than `BuildModule`, but if you like one go for it.

Reply:
I think another one that came up was `Detectable` but honestly I like `BuildModule` the best. I don't know why 🤔